src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -355,6 +361,105 @@ public void validateMedicalInventoryRow(MedicalInventory inventory, List<Medical
+
+		// Get all the lot of the ward movements
+		List<Lot> lotOfMovements = new ArrayList<>(movementWards.stream().map(MovementWard::getLot).toList());
+		// Get all the lot of the movements
+		lotOfMovements.addAll(movementToWards.stream().map(Movement::getLot).toList());
+		// Remove duplicates by converting the list to a set
+		Set<Lot> uniqueLots = new HashSet<>(lotOfMovements);
+		// Convert the set back to a list
+		List<Lot> uniqueLotList = new ArrayList<>(uniqueLots);
+		// Cycle fetched movements to see if they impact inventoryRowSearchList
+		for (Lot lot : uniqueLotList) {
+			String lotCodeOfMovement = lot.getCode();
+			String lotExpiringDate = TimeTools.formatDateTime(lot.getDueDate(), TimeTools.DD_MM_YYYY);
+			String lotInfo = GeneralData.AUTOMATICLOT_IN ? lotExpiringDate : lotCodeOfMovement;
+			Medical medical = lot.getMedical();
+			String medicalDesc = medical.getDescription();
+			Integer medicalCode = medical.getCode();
+			double wardStoreQty = 0.0;
+
+			Optional<MedicalWard> optMedicalWard = movWardBrowserManager.getMedicalsWard(inventory.getWard(), medical.getCode(), false).stream()
+					.filter(m -> m.getLot().getCode().equals(lotCodeOfMovement)).findFirst();
+
+			if (optMedicalWard.isPresent()) {
+				wardStoreQty = optMedicalWard.get().getQty();
+			}
+
+			// Search for the specific Lot and Medical in inventoryRowSearchList (Lot should be enough)
+			Optional<MedicalInventoryRow> matchingRow = inventoryRowSearchList.stream()
+				.filter(row -> row.getLot().getCode().equals(lotCodeOfMovement) && row.getMedical().getCode().equals(medicalCode)).findFirst();
+
+			if (matchingRow.isPresent()) {
+				MedicalInventoryRow medicalInventoryRow = matchingRow.get();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				if (wardStoreQty != theoQty) {
+					lotUpdated = true;
+					medDescriptionForLotUpdated
+						.append("\n")
+						.append(MessageBundle.formatMessage("angal.inventory.theoreticalqtyhavebeenupdatedforsomemedical.detail.fmt.msg",
+							medicalDesc, lotInfo, theoQty, wardStoreQty, wardStoreQty - theoQty));
+				}
+			} else {
+				// TODO: to decide if to give control to the user about this
+				if (!inventoryMedicalsList.contains(medical)) {
+					// New medical
+					medicalAdded = true;
+					medDescriptionForNewMedical
+						.append("\n")
+						.append(MessageBundle.formatMessage("angal.inventory.newmedicalshavebeenfound.detail.fmt.msg",
+							medicalDesc, lotInfo, wardStoreQty));
+				} else {
+					// New Lot
+					lotAdded = true;
+					medDescriptionForNewLot
+						.append("\n")
+						.append(MessageBundle.formatMessage("angal.inventory.newlotshavebeenaddedforsomemedical.detail.fmt.msg",
+							medicalDesc, lotInfo, wardStoreQty));
+				}
+			}
+		}
+		List<OHExceptionMessage> errors = new ArrayList<>();
+		if (lotUpdated) {
+			errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.inventory.validate.btn"),
```

Comment:
why test this again?  you know at line 418 this is true; why not add the `errors.add()` after 422 or better yet get rid of the local variable `medDescriptionForLotUpdated` and just create the exception message in one line.   The same thing is true for `lotAdded` and `medicaladded`.